user-operator/src/test/java/io/strimzi/operator/user/model/KafkaUserModelTest.java

```diff
@@ -297,8 +298,12 @@ public void testGenerateSecretGeneratesPasswordWhenNoUserSecretExists()    {
                         .withKubernetesManagedBy(KafkaUserModel.KAFKA_USER_OPERATOR_NAME)
                         .toMap()));
 
-        assertThat(generatedSecret.getData().keySet(), is(singleton(KafkaUserModel.KEY_PASSWORD)));
-        assertThat(new String(Base64.getDecoder().decode(generatedSecret.getData().get(KafkaUserModel.KEY_PASSWORD))), is("aaaaaaaaaa"));
+        assertThat(generatedSecret.getData().keySet(), is(new HashSet<>(Arrays.asList(KafkaUserModel.KEY_PASSWORD, KafkaUserModel.KEY_SASL_JAAS_CONFIG))));
+
+        String password = "aaaaaaaaaa";
+        assertThat(new String(Base64.getDecoder().decode(generatedSecret.getData().get(KafkaUserModel.KEY_PASSWORD))), is(password));
+        assertThat(new String(Base64.getDecoder().decode(generatedSecret.getData().get(KafkaUserModel.KEY_SASL_JAAS_CONFIG))),
+                is(KafkaUserModel.getSaslJsonConfig(ResourceUtils.NAME, password)));
```

Comment:
I think you are just testing that the base64 encoding (on Kafka user creation) and the decoding (in the assert) are working not that it returns the JAAS string you would expect. You should also have the expected string somewhere to do the assertation that it's corrected.